go-controller/pkg/ovn/subnetallocator/allocator.go

```diff
@@ -5,23 +5,61 @@ import (
 	"net"
 	"sync"
 
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	utilnet "k8s.io/utils/net"
 )
 
 var ErrSubnetAllocatorFull = fmt.Errorf("no subnets available.")
 
-type SubnetAllocator struct {
+type SubnetAllocator interface {
+	AddNetworkRange(network *net.IPNet, hostSubnetLen int) error
+	MarkAllocatedNetworks(string, ...*net.IPNet) error
+	// Usage returns the number of available and used v4 subnets, and
+	// the number of available and used v6 subnets
+	Usage() (uint64, uint64, uint64, uint64)
+	AllocateNetworks(string) ([]*net.IPNet, error)
+	AllocateIPv4Network(string) (*net.IPNet, error)
+	AllocateIPv6Network(string) (*net.IPNet, error)
+	// ReleaseNetworks releases the given networks if they are owned by the
+	// given owner
+	ReleaseNetworks(string, ...*net.IPNet) error
+	// ReleaseAllNetworks releases all networks owned by the given owner
+	ReleaseAllNetworks(string)
+}
+
+type BaseSubnetAllocator struct {
 	sync.Mutex
 
 	v4ranges []*subnetAllocatorRange
 	v6ranges []*subnetAllocatorRange
 }
 
-func NewSubnetAllocator() *SubnetAllocator {
-	return &SubnetAllocator{}
+var _ SubnetAllocator = &BaseSubnetAllocator{}
+
+func NewSubnetAllocator() *BaseSubnetAllocator {
```

Comment:
nit: should we return a `SubnetAllocator` here instead of a `BaseSubnetAllocator`?

Reply:
Done.